### PR TITLE
ajv-keywords 

### PR DESCRIPTION
**Pull request by dgilperez**
Hi! 

I was playing with extra keywords and transformations on the schemas by adding `ajv-keywords` dependency. 

What do you think? Would you consider adding this as is, or should this PR require any changes?

### CODE REVIEW

File: tests/index.js [hunk 1]
```diff
@@ -257,6 +258,23 @@ describe("blocks :: init with defaults", async assert => {
       },
     })
 
+    assert({
+      given: "form encoded with custom keywords",
+      should: "validate input",
```
Comment (andreidmt) [inline tests/index.js:263]: I would be more descriptive here (prob applies for my other tests as well), something like:

```
should: "transform and validate data inside schema with added functionalities from ajv-keywords"
```

File: src/plugins/router.js [hunk 2]
```diff
@@ -46,6 +47,8 @@ module.exports = {
       ],
     })
 
+    addKeywords(ajv)
```
Comment (andreidmt) [inline src/plugins/router.js:50]: `addKeywords` (and also `addFormats`) should have a minimal set of defaults and allow users to add more from `.env`, but this is for another time.

At first glance at [ajv-keywords](https://github.com/ajv-validator/ajv-keywords#contents), I would go with `transform` and `dynamicDefaults`.

**Review by andreidmt — APPROVED** [review]
Looks good, `ajv-keywords` was on the radar. Added a few comments but it's good to merge. :rocket: 

Thank you for also adding tests.

**Comment by dgilperez** [comment]
Bad news, I tested this on my local and it's not working. Error I saw: 

```
Error: Cannot find module 'ajv/dist/compile/codegen'
```

I tracked down to npm version 6 vs 7 issue (at least it was [the solution here](https://github.com/serverless/serverless/issues/8727)) :( Upgrading npm also breaks compatibility with other projects, so I cut losses and stopped. This is going to loose some traction until either we can upgrade npm or this has backwards compatibility.

Also, I learnt that we can achieve the same without `ajv-keywords`: 

```
                {
                  "type": "string",
                  "pattern": "^(activity|itinerary)/.*"
                },
```

Re merging, this is probably a blocking issue though - although npm version support is not stated in the README, it is working on 6 now.

**Comment by andreidmt** [comment]
Wierd, did a clone of your fork and managed to run the tests successfully. Using node 12.20 and npm 7.5.2

```
npm run setup && npm test
```

Also did an `npm run update` to get everything (ajv@7.1.1) up to date just in case.

**Comment by dgilperez** [comment]
Yes, it works for me with those versions, but upgrading our current npm 6
to 7 breaks some other dependencies on the main project :( I managed to
update some but could not update all. :shrug:

>

**Comment by andreidmt** [comment]
Gotcha, will hold the merge then until you upgrade. If there are issues with other asd packages, tell me and ill take a look.

**Comment by dgilperez** [comment]
Closed in favour of #5

**Pull request closed without merging**